app/assets/javascripts/discourse/app/widgets/post.js

```diff
@@ -808,9 +808,28 @@ createWidget("post-notice", {
 
   html(attrs) {
     if (attrs.notice.type === "custom") {
+      let createdByHTML = "";
+      if (attrs.noticeCreatedByUser) {
+        const createdByName =
+          this.siteSettings.display_name_on_posts &&
+          prioritizeNameInUx(attrs.noticeCreatedByUser.name)
+            ? attrs.noticeCreatedByUser.name
+            : attrs.noticeCreatedByUser.username;
+        createdByHTML = i18n("post.notice.custom_created_by", {
+          userLinkHTML: `<a
+                  class="trigger-user-card"
+                  data-user-card="${attrs.noticeCreatedByUser.username}"
+                  title=${createdByName}
+                  aria-hidden="false"
+                  role="listitem"
+                >${createdByName}</a>`,
+        });
+      }
       return [
         iconNode("user-shield"),
-        new RawHtml({ html: `<div>${attrs.notice.cooked}</div>` }),
+        new RawHtml({
+          html: `<div class="post-notice-message">${attrs.notice.cooked} ${createdByHTML}</div>`,
```

Comment:
There appears to be an XSS problem here with the name. For example, when the site is configured to use name, I can craft my name to insert an `iframe` here:

<img width="659" alt="Screenshot 2025-01-23 at 1 24 48 PM" src="https://github.com/user-attachments/assets/266010aa-b0dd-4929-a030-09c3eac723fa" />

Reply:
![Uploading image.png…]()

Reply:
@Drenmi fixed it:

![image](https://github.com/user-attachments/assets/631c7d6f-2ed1-4740-939c-cfda4f7db4e5)